patches/tModLoader/Terraria/ModLoader/NPCLoader.cs

```diff
@@ -1218,5 +1218,22 @@ internal static void DrawBehind(NPC npc, int index)
 				}
 			}
 		}
+
+		private delegate bool DelegateNeedSaving(NPC npc);
+		private static HookList HookNeedSaving = AddHook<DelegateNeedSaving>(g => g.NeedSaving);
+
+		public static bool SavesAndLoads(NPC npc) {
+			if (npc.townNPC && npc.type != NPCID.TravellingMerchant)
+				return false;
+			if (NPCID.Sets.SavesAndLoads[npc.type] || (npc.ModNPC?.NeedSaving() == true))
```

Comment:
need to consider the usage of this set, what's the intended use for it? Should modders use it at all? Should it be fully deprecated and calls replaced with `NPCLoader.SavesAndLoads`?

Reply:
I didn't want to change that but yes it seems fair to remove the access to this array. It's only useful because vanilla uses it honestly, so it's as you want